Document openExternalLinkInNewTab config.settings (#5678)
plone/volto: docs/source/configuration/settings-reference.md

    [See an explanation of character limits in URLs](https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers/417184#417184).
    Please test this setting properly before enabling in a production site.

openExternalLinkInNewTab 
    A Boolean, `false` by default.

>     If openExternalLinkInNewTab is true and the URL is not an internal one, or if the target is '_blank', the openLinkInNewTab will be true, indicating that the link should be opened in a new tab.

Reviewer: I'm not sure whether this is how it actually works. I'd have to research the implementation in the code to be certain. My suggestion is only for English grammar and MyST syntax.

```suggestion
    If either `openExternalLinkInNewTab` is `true` and the URL is external, or if `target` is `_blank`, then `openLinkInNewTab` will be `true`, indicating that the link, when clicked, will open in a new tab.
```

    [See an explanation of character limits in URLs](https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers/417184#417184).
    Please test this setting properly before enabling in a production site.

openExternalLinkInNewTab 
    A Boolean, `false` by default.

>     openExternalLinkInNewTab allows the user to control whether links are opened in a new tab or the current tab, based on their settings.

Reviewer 2: @ichim-david Well, I think I was jumping to conclusions. I think UniversalLink _should_ use this setting by default, but I don't think it does yet.

>     openExternalLinkInNewTab allows the user to control whether links are opened in a new tab or the current tab, based on their settings.

Reviewer: I have only MyST syntax and Plone style guide suggestions.

Given we are not sure about the actual behavior, I'm not sure how to proceed.

>     openExternalLinkInNewTab allows the user to control whether links are opened in a new tab or the current tab, based on their settings.

Reviewer 3: > @ichim-david Well, I think I was jumping to conclusions. I think UniversalLink _should_ use this setting by default, but I don't think it does yet.

@davisagli I don't know how you read my comment if you've read it from the email notification or by looking at it from the github website.
What I wanted your feedback was to the suggestion I had for the short summary of what this setting does which I wrote as a suggestion that can be committed here https://github.com/plone/volto/pull/5678#discussion_r1473200937

I want to try to convey that this setting is used by the UniversalLink but we suggest that developers also use it in their components if they want to control the behavior of the target of external links and I wanted to hear your thoughts about this, if you would have this statement added or not.

>     openExternalLinkInNewTab allows the user to control whether links are opened in a new tab or the current tab, based on their settings.

Reviewer 2: @ichim-david Yeah, I saw your suggestion here. But UniversalLink doesn't currently use this setting; it just takes a openLinkInNewTab prop. So your suggestion describes how I think it should work, but not how it currently works.

>     openExternalLinkInNewTab allows the user to control whether links are opened in a new tab or the current tab, based on their settings.

Reviewer 2: (Or, possibly, I'm missing something.)

>     openExternalLinkInNewTab allows the user to control whether links are opened in a new tab or the current tab, based on their settings.

Reviewer 3: > @ichim-david Yeah, I saw your suggestion here. But UniversalLink doesn't currently use this setting; it just takes a openLinkInNewTab prop. So your suggestion describes how I think it should work, but not how it currently works.

@davisagli https://github.com/plone/volto/pull/4379/files#diff-20b361f846b3f22542a6c5784e47d22c783206e12623d183d16dc7695d414a13R7

>     openExternalLinkInNewTab allows the user to control whether links are opened in a new tab or the current tab, based on their settings.

Reviewer 2: @ichim-david That is using the setting outside of UniversalLink and passing it in. It's not used directly / automatically by UniversalLink like your suggestion here says.

>     openExternalLinkInNewTab allows the user to control whether links are opened in a new tab or the current tab, based on their settings.

Reviewer 4: IMO, the fix to issue [openExternalLinkInNewTab](https://github.com/plone/volto/issues/5693) and this PR should be combined together. This can't be merged without it.
